### DISCUSSION TITLE
How to create a console based worker cluster node

### DISCUSSION THREAD

**Discussion by johnwc** [elsa-workflows/elsa-core#3157] [Q&A]
The current example for console hello world is executing a single code based workflow. We will be creating several containers in kubernetes to host our worker nodes, along with one container hosting the dashboard. How do we create a console node for clustering without creating a asp.net core web app. Or is this not possible for cluster nodes?

**Comment by sfmskywalker** [comment]
Depending on what capabilities you need from the workflow worker nodes, the best way might be to scaffold a new project using the WorkerService template. This generates a project that only runs a background service. You can delete that, and simply start adding Elsa services to the service collection, as you normally would do in an ASP.NET Core project (without ASP.NET-related code such as endpoints and controllers).

Setting up a worker service project instantiates an `IHost` that keeps your application running and can host other hosted services, such as message bus consumers, Quartz jobs, Hangfire jobs, etc.

**Reply by johnwc** [reply]
What template are you referring to? Is there a link to it?
